### ISSUE TITLE
asVector prop to render FeatureCollection using GeoJSONLayer

### ISSUE THREAD

**Issue by kylebarron** [UnfoldedInc/earthengine-layers#34]
Calling `FeatureCollection.getInfo()` returns a GeoJSON `FeatureCollection`, but downloading happens within the EE API, so streaming data to Deck.gl would be harder.

It's possible to use `getDownloadURL` to get a direct download URL to be used with streaming:
```py
url = FeatureCollection.getDownloadURL('json', selectors=['.geo'])
```
`.geo` is necessary to download vector geometries. When selectors is only `.geo`, no `properties` are included. You can find all property names of the table with `FeatureCollection.propertyNames().getInfo()`, and those could be added to the `selectors` list.

It's then possible to use `JSONLoader.loadInBatches` to stream data into a GeoJSON layer.

Geometries are often very precise. The GeoJSON of the 50 states from `TIGER/2016/States` is 34MB, which is most likely more precise than necessary for most applications. There could be a helper function in loaders.gl to simplify geometries

**Comment by ibgreen** [comment]
Decision: Add a `limit` or `featureLimit` prop, set a reasonable default, apply to `ee.FeatureCollection` before generating URL.

**Comment by kylebarron** [comment]
It would also be possible to use [`FeatureCollection.filterBounds`](https://developers.google.com/earth-engine/api_docs#ee.featurecollection.filterbounds) to download _tiled_ GeoJSON. I.e. you could call `filterBounds` with the bounds of the mercator tile, then get a download Url for just that tile. 

If you're working with a vector dataset covering a large area, it could help to minimize the amount downloaded.

**Comment by ibgreen** [comment]
Neat idea. It could also help us do tile based analytics. 

It may be heavy on the backend though, let's do some investigation first.

We'd also rely on cut polygons/features being stitched together correctly and making sure separate pieces maintaining a common identity for picking / highlighting etc.

**Comment by kylebarron** [comment]
I think this should be an optional enhancement for the future. The docs say

> Shortcut to filter a collection by intersection with geometry. Items in the collection with a footprint that fails to intersect the given geometry will be excluded.

So this isn't a _clip_ per se. It's more like searching an Rtree index for a specific tile, which includes false positives.

**Comment by kylebarron** [comment]
Closed with #58
